### PR TITLE
feat: auction modal

### PR DESCRIPTION
**Pull request by NicoSantangelo**
Auction modal flow:

1) If you enter `/auction` for the first time and you don't have the contract authorized, you'll see a button redirecting you to Settings.
2) If you enter `/auction` for the first time and you do have the contract authorized, you'll have a button to start that dismisses the modal
3) If you enter `/auction` for the second time and you didn't interact with the modal you'll see 1 or 2 depending on the initial condition
4) If you enter `/auction` having accepted 2) you won't see anything

It also refactors the base modal a bit to keep it consistent, remove unused code and styles and working on all resolutions.

Closes #632 
Continues #626

### CODE REVIEW

File: src/Translation/locales/es.json [hunk 1]
```diff
@@ -103,6 +103,15 @@
     "your_estate": "Su propiedad",
     "your_parcel": "Tu parcela"
   },
+  "auction_modal": {
+    "authorize_contract": "Contrato de autorización",
+    "authorized_explanation": "Estás listo para comenzar! ",
```
Comment (nachomazzara) [inline src/Translation/locales/es.json]: Oops., It is different from English translation and also there is an extra space at the end `"Estás listo para comenzar! "'

File: src/Translation/locales/es.json [hunk 2]
```diff
@@ -103,6 +103,15 @@
     "your_estate": "Su propiedad",
     "your_parcel": "Tu parcela"
   },
+  "auction_modal": {
+    "authorize_contract": "Contrato de autorización",
```
Comment (nachomazzara) [inline src/Translation/locales/es.json]: `Autorizar contrato`

File: src/Translation/locales/es.json [hunk 3]
```diff
@@ -103,6 +103,15 @@
...
+    "authorized_explanation": "Estás listo para comenzar! ",
+    "blog_post": "Entrada en el blog",
+    "first_time_explanation":
+      "Para participar en esta nueva subasta, deberá autorizar el nuevo contrato {contract_link}. ",
+    "lets_go": "¡Vamonos!",
```
Comment (nachomazzara) [inline src/Translation/locales/es.json]: where? hahahahahahahha

File: src/Translation/locales/es.json [hunk 4]
```diff
@@ -103,6 +103,15 @@
...
+    "title": "Decentraland Subasta"
```
Comment (nachomazzara) [inline src/Translation/locales/es.json]: `Subasta de Decentraland`

File: src/Translation/locales/es.json [hunk 5]
```diff
@@ -405,6 +414,18 @@
         "Puedes usar navegadores móviles como {coinbase_link} o {imtoken_link}."
     }
   },
+  "terms_modal": {
+    "also_agree":
+      "También acepta nuestro {privacy_policy_link}, que describe cómo procesamos su información.",
+    "by_agreeing":
+      "Al seleccionar \"Acepto\" a continuación, acepta nuestro {terms_of_service_link}.",
+    "explanation":
+      "Este dApp interactúa con un intercambio descentralizado que se ejecuta en la cadena de bloques Ethereum.",
```
Comment (nachomazzara) [inline src/Translation/locales/es.json]: `***Esta***...cadena de bloques ***de*** ....`

Maybe we should keep `intercambio` as `exchange`

File: src/Translation/locales/es.json [hunk 6]
```diff
@@ -405,6 +414,18 @@
...
+    "i_agree": "Estoy de acuerdo",
+    "privacy_policy": "Política de privacidad",
+    "terms_of_service": "Términos de servicio",
+    "title": "Bienvenido a Decentraland's LAND Marketplace"
```
Comment (nachomazzara) [inline src/Translation/locales/es.json]: `Bienvenido al Mercado de LAND de Decentaland` ?

File: webapp/src/components/AuctionPage/AuctionPage.js [hunk 7]
```diff
@@ -25,6 +28,25 @@ export default class AuctionPage extends React.PureComponent {
...
+  componentWillReceiveProps(nextProps) {
+    this.showAuctionModal(nextProps)
+  }
+
+  showAuctionModal(props) {
+    const { authorization, onShowAuctionModal } = props
+
+    if (
+      !hasSeenAuctionModal() ||
+      (authorization && !isAuthorized(authorization))
```
Comment (nachomazzara) [inline webapp/src/components/AuctionPage/AuctionPage.js]: What do you think of moving`(authorization && !isAuthorized(authorization))` into a variable like

```javascript
const isAuctionContractAuthorized = authorization && !isAuthorized(authorization)
```

File: webapp/src/components/BuyEstatePage/BuyEstatePage.js [hunk 8]
```diff
@@ -116,7 +113,11 @@ export default class BuyEstatePage extends React.PureComponent {
     const { isConnected, isLoading } = this.props
 
     if (isLoading) {
-      return <Loading />
+      return (
```
Comment (nachomazzara) [inline webapp/src/components/BuyEstatePage/BuyEstatePage.js]: Hmmm, Do you remember why you need to change it?

It would be great to just use `<Loading />` to show the spinner to avoid code duplication between components

File: webapp/src/components/ContractLink/ContractLink.js [hunk 9]
```diff
@@ -0,0 +1,21 @@
+import React from 'react'
```
Comment (nachomazzara) [inline webapp/src/components/ContractLink/ContractLink.js]: I like this component and I also have the same idea. But If I am not wrong, this component would be useful to have it at `decentraland-dapps`.

It is not part of the PR but we can create an issue to move it there

File: webapp/src/components/Modal/AuctionModal/AuctionModal.container.js [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+import { navigateTo } from '@dapps/modules/location/actions'
+import { getAddress } from 'modules/wallet/selectors'
+import { getData as getAuthorizations } from 'modules/authorization/selectors'
+
+import AuctionModal from './AuctionModal'
+
+const mapState = state => {
+  const address = getAddress(state)
+  const authorization = getAuthorizations(state)[address]
+
```
Comment (nachomazzara) [inline webapp/src/components/Modal/AuctionModal/AuctionModal.container.js]: If we don't have a case where we care about authoriazations of not the current address. What do you think of moving `getAddress(state)` as part of the` getAuthorizations` selector?

We could avoid the need of doing:

```javascript
const address = getAddress(state)
const authorization = getAuthorizations(state)[address]
```

And do just 
```javascript
const authorization = getAuthorizations(state)
```

File: webapp/src/components/Modal/AuctionModal/AuctionModal.js [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+  static propTypes = {
+    ...BaseModal.propTypes,
+    authorization: authorizationType,
+    onAuthorize: PropTypes.func.isRequired
+  }
+
+  handleOnAuthorize = () => {
+    const { onClose, onAuthorize } = this.props
+    onClose()
+    onAuthorize()
```
Comment (nachomazzara) [inline webapp/src/components/Modal/AuctionModal/AuctionModal.js]: Is it ok to close and authorize? I'm wondering if changing the order has an effect

File: src/Translation/locales/es.json (hunk 3 above)
Comment (NicoSantangelo) [reply src/Translation/locales/es.json]: HAHAH

File: webapp/src/components/ContractLink/ContractLink.js (hunk 9 above)
Comment (NicoSantangelo) [reply webapp/src/components/ContractLink/ContractLink.js]: I like the idea, @cazala what do you think?

File: webapp/src/components/Modal/AuctionModal/AuctionModal.js (hunk 11 above)
Comment (NicoSantangelo) [reply webapp/src/components/Modal/AuctionModal/AuctionModal.js]: it should be the same, the onClose only changes a boolean in the global redux state. I wanted to close it first because `onAuthorize` is a redirect. So you close and _then_ see something else.
In practice it's so fast that it's the same thing

File: webapp/src/components/Modal/AuctionModal/AuctionModal.container.js (hunk 10 above)
Comment (NicoSantangelo) [reply webapp/src/components/Modal/AuctionModal/AuctionModal.container.js]: Hmm, we may need authorisations for another address, but I guess it's not that common. I'm not sure about coupling a method name like `getAuthorizations` with `getAddress` but I see your point.

I cast @cazala 🧙‍♂️. What do you think?

File: webapp/src/components/BuyEstatePage/BuyEstatePage.js (hunk 8 above)
Comment (NicoSantangelo) [reply webapp/src/components/BuyEstatePage/BuyEstatePage.js]: My reasoning behind removing this is that this component looked like accidental duplication. It's only function was enabling `<Loader />` for `display: flex` parents.

That said, I can see that we're repeating:

`<div><Loader /></div>`

in quite a few places. I'm not sure if we should abstract this or not, if we do, I'd do it top level, but Loading sounds weird when we do use `Loader` in a couple of places. Once again @cazala  what do you think?

File: webapp/src/components/ContractLink/ContractLink.js (hunk 9 above)
Comment (cazala) [reply webapp/src/components/ContractLink/ContractLink.js]: 👍 

There are a bunch of Etherscan related components that we have and could live as a component or family of components like BlockDate, EtherscanLink and this one

File: webapp/src/components/Modal/AuctionModal/AuctionModal.container.js (hunk 10 above)
Comment (NicoSantangelo) [reply webapp/src/components/Modal/AuctionModal/AuctionModal.container.js]: Ok! I checked and we're already doing this in a couple of places so I'll go with that!

File: webapp/src/components/ContractLink/ContractLink.js (hunk 9 above)
Comment (NicoSantangelo) [reply webapp/src/components/ContractLink/ContractLink.js]: https://github.com/decentraland/decentraland-dapps/issues/24

**Comment by NicoSantangelo** [comment]
Updated!

**Comment by nachomazzara** [comment]
We can merge it after rebase (L
